packages/xo-web/src/xo-app/proxies/deploy-proxy.js

```diff
@@ -249,20 +255,31 @@ const deployProxy = async ({ proxy } = {}) => {
     )
   }
 
-  const header = (
-    <span>
-      <Icon icon='proxy' />{' '}
-      {isRedeployMode ? _('redeployProxy') : _('deployProxy')}
-    </span>
-  )
-
+  const title = isRedeployMode ? _('redeployProxy') : _('deployProxy')
   if (license === undefined) {
-    return alert(
-      header,
-      <div className='text-muted'>
-        <Icon icon='info' /> {_('noLicenseAvailable')}
-      </div>
-    )
+    const value = 'trial'
+    const choice = await chooseAction({
+      body: (
+        <div className='text-muted'>
+          <Icon icon='info' /> {_('noLicenseAvailable')}
+        </div>
+      ),
+      buttons: [
+        {
+          btnStyle: 'success',
+          icon: 'trial',
+          label: _('trialStartButton'),
+          value,
+        },
+      ],
+      icon: 'proxy',
+      title,
+    }).catch(noop)
+    if (choice !== value) {
```

Comment:
Why do you not inline it and create the trial licence if `choice === value`?

Reply:
Because we should also continue the deployment process. See https://github.com/vatesfr/xen-orchestra/pull/5022/files#diff-0e624c10572d77a9ee7f7d495e149ae9R285.